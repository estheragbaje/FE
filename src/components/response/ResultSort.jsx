import React, { Component } from 'react';
import styled from 'styled-components';
import Chart from './Chart';
import { extraLargeSpace } from '../../styles/variables';


class ResultSort extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { questions } = this.props;
    return (
      <ResultContainer>
        <Chart data={questions} />
      </ResultContainer>
    );
  }
}

const ResultContainer = styled.div`
display: flex;
flex-direction: column;
justify-content: center;
margin: 0 0 0 ${extraLargeSpace};
`;

export default ResultSort;
