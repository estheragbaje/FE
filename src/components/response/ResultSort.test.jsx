import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { ResultSort } from './ResultSort';


const mockProps = {
  title: 'test title',
  id: '234455-2345',
  questions: [
    {
      id: 1,
      type: 'text',
      question: 'What do you say about my overall usefulness in the team for the past 4 months?',
      feedbacks: [
        {
          id: 1,
          rating: '7',
          comment: null,
        },
        {
          id: 2,
          rating: null,
          comment: "It's been quite incredible",
        },
      ],
    },
    {
      id: 2,
      type: 'text',
      question: 'What do you say about my overall usefulness in the team for the past 4 months?',
      feedbacks: [
        {
          id: 1,
          rating: null,
          comment: "It's been quite incredible",
        },
        {
          id: 2,
          rating: 3,
          comment: null,
        },
      ],
    },
  ],
};

let wrapper;
beforeAll(() => {
  // eslint-disable-next-line react/jsx-props-no-spreading
  wrapper = shallow(<ResultSort {...mockProps} />);
});
describe('<ResultSort />', () => {
  it('renders the ResultSort component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
