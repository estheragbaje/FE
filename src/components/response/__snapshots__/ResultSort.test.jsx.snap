// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultSort /> renders the ResultSort component correctly 1`] = `
<styled.div>
  <Chart
    data={
      Array [
        Object {
          "feedbacks": Array [
            Object {
              "comment": null,
              "id": 1,
              "rating": "7",
            },
            Object {
              "comment": "It's been quite incredible",
              "id": 2,
              "rating": null,
            },
          ],
          "id": 1,
          "question": "What do you say about my overall usefulness in the team for the past 4 months?",
          "type": "text",
        },
        Object {
          "feedbacks": Array [
            Object {
              "comment": "It's been quite incredible",
              "id": 1,
              "rating": null,
            },
            Object {
              "comment": null,
              "id": 2,
              "rating": 3,
            },
          ],
          "id": 2,
          "question": "What do you say about my overall usefulness in the team for the past 4 months?",
          "type": "text",
        },
      ]
    }
  />
</styled.div>
`;
